sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -141,10 +146,127 @@ AccessToken TokenCredentialImpl::GetToken(
 }
 
 namespace {
-[[noreturn]] void ThrowJsonPropertyError(std::string const& propertyName)
+std::string const ParseTokenLogPrefix = "TokenCredentialImpl::ParseToken(): ";
+
+std::string TokenAsDiagnosticString(
+    json const& jsonObject,
+    std::string const& accessTokenPropertyName,
+    std::string const& expiresInPropertyName,
+    std::string const& expiresOnPropertyName)
 {
+  std::stringstream ss;
+  ss << "Token JSON";
+
+  if (!jsonObject.is_object())
+  {
+    ss << " is not an object (value='" << jsonObject.dump() << "')";
+  }
+  else
+  {
+    ss << ": Access token property ('" << accessTokenPropertyName << "') ";
+    if (!jsonObject.contains(accessTokenPropertyName))
+    {
+      ss << "is NOT present";
+    }
+    else
+    {
+      auto const& accessTokenProperty = jsonObject[accessTokenPropertyName];
+      if (!accessTokenProperty.is_string())
+      {
+        ss << "is NOT a string (value='" << accessTokenProperty.dump() << "')";
+      }
+      else
+      {
+        ss << "is string (length=" << accessTokenProperty.get<std::string>().length() << ")";
+      }
+    }
+
+    for (auto const& p : {
+             std::pair<char const*, std::string const*>{"relative", &expiresInPropertyName},
+             std::pair<char const*, std::string const*>{"absolute", &expiresOnPropertyName},
+         })
+    {
+      ss << ", " << p.first << " expiration property ('" << *p.second << "') ";
+      if (!jsonObject.contains(*p.second))
+      {
+        ss << "is NOT present";
+      }
+      else
+      {
+        ss << "is present (value='" << jsonObject[*p.second].dump() << "')";
+      }
+    }
+
+    std::map<std::string, json> otherProperties;
+    for (auto const& property : jsonObject.items())
+    {
+      if (property.key() != accessTokenPropertyName && property.key() != expiresInPropertyName
```

Comment:
Also don't display "accesstoken" || "access_token"

Reply:
Addressed, but in the other way: please see the new commit (unit tests have all sorts of example outputs).
We would never print any string value for access token property, if it was found.
If it is an unknown property, we would only print the size os the string, unless it belongs to a subset of known strings (datetimes, integers). There is more, I think the unit test would explain better.